core/shared/platform/zephyr/zephyr_file.c

```diff
@@ -711,22 +713,24 @@ os_renameat(os_file_handle old_handle, const char *old_path,
     char abs_old_path[MAX_FILE_NAME + 1];
     char abs_new_path[MAX_FILE_NAME + 1];
 
+    GET_FILE_SYSTEM_DESCRIPTOR(old_handle->fd, ptr);
+
+    char *path = strdup(new_path);
+    if (path == NULL) {
+        return __WASI_ENOMEM;
```

Comment:
In before, if `strdup()` returns NULL, `ptr->path` will be set to `old_path` instead of maintaining original value. But in after, it's not. So, is it on purpose?